detection_rules/main.py

```diff
@@ -39,33 +40,38 @@ def root():
 @click.option('--rule-type', '-t', type=click.Choice(CurrentSchema.RULE_TYPES), help='Type of rule to create')
 def create_rule(path, config, required_only, rule_type):
     """Create a detection rule."""
-    config = load_dump(config) if config else {}
+    contents = load_rule_contents(config, {})
     try:
-        return Rule.build(path, rule_type=rule_type, required_only=required_only, save=True, **config)
+        return Rule.build(path, rule_type=rule_type, required_only=required_only, save=True, **contents)
     finally:
         rule_loader.reset()
 
 
-@root.command('load-from-file')
+@root.command('import-rules')
 @click.argument('infile', type=click.Path(dir_okay=False, exists=True), nargs=-1, required=False)
+@click.option('--multi-file', '-m', type=click.Path(dir_okay=False, exists=True),
+              help='File with multiple exported rules')
 @click.option('--directory', '-d', type=click.Path(file_okay=False, exists=True), help='Load files from a directory')
-def load_from_file(infile, directory):
-    """Load rules from file(s)."""
-    if infile:
-        for rule_file in infile:
-            rule_path = os.path.join(RULES_DIR, os.path.basename(rule_file))
-            rule = Rule(rule_path, load_dump(rule_file))
-            rule.save(as_rule=True, verbose=True)
-    elif directory:
-        for rule_file in glob.glob(os.path.join(directory, '**', '*.*'), recursive=True):
-            try:
-                rule_path = os.path.join(RULES_DIR, os.path.basename(rule_file))
-                rule = Rule(rule_path, load_dump(rule_file))
-                rule.save(as_rule=True, verbose=True)
-            except ValueError:
-                click.echo('Unable to load file: {}'.format(rule_file))
-    else:
-        click.echo('No files specified!')
+def import_rules(infile, multi_file, directory):
+    """Import rules from json, toml, or Kibana exported rule file(s)."""
+    rule_files = glob.glob(os.path.join(directory, '**', '*.*'), recursive=True) if directory else []
+    rule_files = sorted(list(set(rule_files + list(infile))))
+    rule_contents = [load_rule_contents(rf) for rf in rule_files]
+
+    if multi_file:
+        rule_contents.extend(load_multi_rule_contents(multi_file))
+
+    if not rule_contents:
+        click.echo('Must specify at least one file!')
+
+    def name_to_filename(name):
+        return re.sub(r'[^_a-z0-9]+', '_', name.strip().lower()).strip('_') + '.toml'
+
+    for contents in [c for c in rule_contents if c]:
+        base_path = contents.get('name') or contents.get('rule', {}).get('name')
+        base_path = name_to_filename(base_path) if base_path else base_path
+        rule_path = os.path.join(RULES_DIR, base_path) if base_path else None
+        Rule.build(rule_path, required_only=True, save=True, verbose=True, **contents)
```

Comment:
This is the easiest way to reconcile diffs in downloaded rules (diffs in stack versions) with a present schema since it will drop unknown fields and prompt for required fields.